tc/sgx/trusted_worker_manager/enclave/work_order_processor.cpp

```diff
@@ -349,13 +348,23 @@ namespace tcf {
             if (i < out_data_size) {
                 tcf::WorkOrderDataHandler& out_data = data_items_out.at(i);
                 out_data.workorder_data.decrypted_data = data.decrypted_data;
-                out_data.ComputeHashString(session_key_iv);
+                out_data.ComputeHashString();
             } else {
-                tcf::WorkOrderDataHandler out_data(data, encrypted_session_key);
-                out_data.ComputeHashString(session_key_iv);
+                // Assign data_encryption_key, iv, data_iv and encrypted_data_encryption_key
+                // params with corresponging values from inData
+                ByteArray encryption_key = data_items_in[0].GetEncryptionKey();
```

Comment:
key can be different per daya item shouldn't data_items_in[0]  -> data_items_in[i]

Reply:
Done